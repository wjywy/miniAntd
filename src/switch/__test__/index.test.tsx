import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import Switch from '../';

describe('Switch', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(
      <div data-testid="switch">
        <Switch></Switch>
      </div>,
    );
    expect(getByTestId('switch')).toMatchSnapshot();
  });

  it('should be false and then be true', () => {
    let f = false;
    const mockCallback = (flag: boolean) => {
      f = flag; //预期是通过回调函数拿到返回值
    };
    const { getByTestId } = render(
      <Switch data-testid="switch" onChange={mockCallback} />,
    );
    const sw = getByTestId('switch');
    fireEvent.click(sw);
    expect(f).toBeFalsy(); //第一次点击状态应该变为false

    fireEvent.click(sw);
    expect(f).toBeTruthy(); //第二次点击状态应该变为true
  });
});
