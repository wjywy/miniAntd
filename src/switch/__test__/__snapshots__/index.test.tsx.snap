// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div
  data-testid="switch"
>
  <div
    class="switch l-switch-md"
    style="background: rgb(24, 144, 255);"
  >
    <div
      class="circle"
    />
  </div>
</div>
`;

exports[`Switch should render correctly 1`] = `
<div
  data-testid="switch"
>
  <div
    class="switch l-switch-md"
    style="background: rgb(24, 144, 255);"
  >
    <div
      class="circle"
    />
  </div>
</div>
`;
